### PR TITLE
SALTO-3183 - Zendesk Adapter - Validate that the user is not attempting to delete the default group

### PR DESCRIPTION
**Pull request by idoser96**
Added a change validator that prevents the user from attempting to delete the group that is the default group
Also deploy group removal changes last, in case the default group was changed, and then that group was deleted

---

None

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/zendesk-adapter/src/change_validator.ts [hunk 1]
```diff
@@ -111,6 +111,7 @@ export default ({
     guideOrderDeletionValidator,
     // ******************************
     macroActionsTicketFieldDeactivationValidator,
+    defaultGroupDeletion,
```
Comment (netama) [inline packages/zendesk-adapter/src/change_validator.ts:114]: let's add this before the Guide comments?

File: packages/zendesk-adapter/src/change_validators/default_group_deletion.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import {
+  ChangeValidator, getChangeData,
+  isInstanceChange, isRemovalChange,
+} from '@salto-io/adapter-api'
+import { GROUP_TYPE_NAME } from '../constants'
+
+export const defaultGroupDeletion: ChangeValidator = async changes => (
```
Comment (netama) [inline packages/zendesk-adapter/src/change_validators/default_group_deletion.ts:22]: please add a docstring

File: packages/zendesk-adapter/src/change_validators/default_group_deletion.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  changes
+    .filter(isInstanceChange).filter(isRemovalChange).map(getChangeData)
+    .filter(instance => instance.elemID.typeName === GROUP_TYPE_NAME)
+    .filter(group => group.value.default === true)
+    .map(group => ({
+      elemID: group.elemID,
+      severity: 'Error',
+      message: 'Default group cannot be deleted',
```
Comment (netama) [inline packages/zendesk-adapter/src/change_validators/default_group_deletion.ts:30]: discussed IRL - let's try to see if we can make deploy that both sets a different default + removes the current default will work

**Review by netama — COMMENTED** [review]
looks good :) discussed IRL a more complex scenario that we should probably support here

File: packages/zendesk-adapter/src/change_validators/default_group_deletion.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import {
+  Change,
+  ChangeValidator, getChangeData, isAdditionOrModificationChange,
+  isInstanceChange, isRemovalChange,
+} from '@salto-io/adapter-api'
+import { GROUP_TYPE_NAME } from '../constants'
+
+// Check if there was any change that marked a new default group
+const isThereNewDefaultGroup = (changes: Change[]): boolean =>
+  changes.filter(isInstanceChange).filter(isAdditionOrModificationChange).map(getChangeData)
```
Comment (netama) [inline packages/zendesk-adapter/src/change_validators/default_group_deletion.ts:25]: is there a need to filter again by `isInstanceChange`?

**Review by netama — APPROVED** [review]
🎉

File: packages/zendesk-adapter/src/change_validators/default_group_deletion.ts (hunk 4 above)
Comment (idoser96) [reply packages/zendesk-adapter/src/change_validators/default_group_deletion.ts:25]: Nope

**Comment by idoser96** [comment]
Closing because at the moment its impossible to edit the default group using Zendesk Api (might be possible in the future)

**Pull request closed without merging**